src/main/scala/mesosphere/marathon/core/externalvolume/ExternalVolumes.scala

```diff
@@ -14,14 +14,15 @@ import org.apache.mesos.Protos.ContainerInfo
 object ExternalVolumes {
   private[this] lazy val providers: ExternalVolumeProviderRegistry = StaticExternalVolumeProviderRegistry
```

Comment:
I want to avoid scope creep here, so feel free to ignore / address in a separate commit.

Does this trait / object look like a pretty classic case of speculative generality to you? The more I look at it, the more confused I am. It seems like `ExternalVolumeProviderRegistry`, and `StaticExternalVolumeProviderRegistry` could just be removed, and turned in to, at worst, a map. And, there's only one volume provide. So it could just be a single method with an assertion.

Reply:
I think this was done having some idea about possible extensions in the future. Added to the description of https://jira.mesosphere.com/browse/MARATHON_EE-1735

Reply:
Yeah, I can understand the motive. It's speculative generality to write code that would possibly allow us to write other extensions in the future. Code is already extensible. We can just change the code. :) All this extra general code is just noise and provides no actual value.

Reply:
Fixed.